core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java

```diff
@@ -161,6 +161,18 @@ default ConnectorTableHandle makeCompatiblePartitioning(ConnectorSession session
         return Optional.empty();
     }
 
+    /**
+     * Return basic table metadata for the specified table handle.
+     * This method is useful when getting full table metadata is expensive.
+     * Engine will try to leverage basic table metadata whenever possible.
+     *
+     * @throws RuntimeException if table handle is no longer valid
+     */
+    default ConnectorTableBasicMetadata getTableBasicMetadata(ConnectorSession session, ConnectorTableHandle table)
```

Comment:
How about calling it `getBasicTableMetadata`? No matter how I turn `getTableBasicMetadata` around in my head, I can't make it parse as anything like "get basic metadata for table."

The more direct `getBasicMetadataForTable` also seems like a good name to me.

I think even `getConnectorTableBasicMetadata` is better than `getTableBasicMetadata`, but it's very verbose, and `get[ReturnType]` is not a very useful function name (but I think it's better than the current version, which is neither the return type nor phrase with similar meaning).

Reply:
Renamed it to `ConnectorTableSchema` after discussion with @martint .